### PR TITLE
[CARBONDATA-2468] addition of column to default sort_column is handled

### PR DESCRIPTION
**Pull request by rahulforallp**
**issue :** default sort_column handling was missing
**solution :**  condition added for default sort_columns

 - [x] Any interfaces changed? NO
 
 - [x] Any backward compatibility impacted?No
 
 - [x] Document update required?NO

 - [x] Testing done==> UT added
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4646/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5801/

File: store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java [hunk 1]
```diff
@@ -481,9 +482,13 @@ private void buildTableSchema(Field[] fields, TableSchemaBuilder tableSchemaBuil
           ColumnSchema columnSchema = tableSchemaBuilder
               .addColumn(new StructField(field.getFieldName(), field.getDataType()),
                   valIndex, isSortColumn > -1);
-          columnSchema.setSortColumn(true);
           if (isSortColumn > -1) {
+            columnSchema.setSortColumn(true);
             sortColumnsSchemaList[isSortColumn] = columnSchema;
+          } else if (sortColumnsList.isEmpty() && columnSchema.isDimensionColumn()) {
```
Comment (kumarvishal09) [inline store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java]: Complex column and it's children should not be considered as sort column, please handle the same

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4877/

File: store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java (hunk 1 above)
Comment (rahulforallp) [reply store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java]: complex check is already handled above.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5830/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4878/

**Comment by brijoobopanna** [comment]
Please fix the build

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4687/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5843/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4889/

**Comment by rahulforallp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4704/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5858/

**Comment by kumarvishal09** [comment]
LGTM

**Pull request closed without merging**